cypress/e2e/jackwang.cy.ts

```diff
@@ -48,26 +46,14 @@ it('73588',()=> {
 it('81012', ()=> {
         cy.visit('https://obstaclecourse.tricentis.com/Obstacles/81012')
         cy.get('#alerttext').invoke('text').then((text) => {
-            const regex = /\$([\d.,]+)/;
-            const matches = text.match(regex);
-            const amount = matches ? matches[1] : null;
-
-            console.log(amount);
-
-            cy.get('#totalamountText').type(amount);
+            const regex = /\$([\d.,]+)/
+            const matches = text.match(regex)
+            const amount = matches ? matches[1] : null
 
+            if (amount != null){
```

Comment:
This is incorrect here. amount is supposed to have a value. It is a problem if amount is null and it should fail the test to report the problem. However you are over doing check and hide the problem.

Reply:
Shouldn't I need to check if the value is null? Is this supposed to be something developers do? Or should I throw this error out in the test script

Reply:
You can throw out error if it is null. The only reason throwing out error is to report failure at a more accurate point. You need to change your mindset to a test engineer and thinking from a real user 's perspective.